app/models/git_repository.rb

```diff
@@ -25,6 +25,27 @@ def self.delete_repo_dir(id, directory_name)
     FileUtils.rm_rf(directory_name)
   end
 
+  # Ping the git repository endpoint to verify that the network connection is still up.
+  # We use this approach for now over Rugged#check_connection because of a segfault:
+  #
+  #   https://github.com/libgit2/rugged/issues/859
+  #
+  def check_connection?
+    require 'net/ping/external'
+
+    # URI library cannot handle git urls, so just convert it to a standard url.
```

Comment:
I just noticed this:  Will this work fine with the colon?  By that I mean, if the URL is like this?

```
git@github.com:org/repo
```

Where a `:` is usually reserved for a port.  I guess I can just check myself...

Reply:
Oh, sorry, I missed the first `sub(':', '/') prior to that.  Nevermind.